counterexamples/cyclotomic_105.lean

```diff
@@ -0,0 +1,120 @@
+
+lemma proper_divisors_105 : nat.proper_divisors 105 = {1, 3, 5, 7, 15, 21, 35} := rfl
+
+end computation
+
+open polynomial
+
+lemma cyclotomic_3 : cyclotomic 3 ℤ = 1 + X + X ^ 2 :=
+begin
+  refine ((eq_cyclotomic_iff (show 0 < 3, by norm_num) _).2 _).symm,
+  rw nat.prime.proper_divisors prime_3,
+  simp only [finset.prod_singleton, cyclotomic_one],
+  ring
+end
+
+lemma cyclotomic_5 : cyclotomic 5 ℤ = 1 + X + X ^ 2 + X ^ 3 + X ^ 4 :=
+begin
+  refine ((eq_cyclotomic_iff (nat.prime.pos prime_5) _).2 _).symm,
+  rw nat.prime.proper_divisors prime_5,
+  simp only [finset.prod_singleton, cyclotomic_one],
+  ring
+end
+
+lemma cyclotomic_7 : cyclotomic 7 ℤ = 1 + X + X ^ 2 + X ^ 3 + X ^ 4 + X ^ 5 + X ^ 6 :=
+begin
+  refine ((eq_cyclotomic_iff (nat.prime.pos prime_7) _).2 _).symm,
+  rw nat.prime.proper_divisors prime_7,
+  simp only [finset.prod_singleton, cyclotomic_one],
+  ring
+end
+
+lemma cyclotomic_15 : cyclotomic 15 ℤ = 1 - X + X ^ 3 - X ^ 4 + X ^ 5 - X ^ 7 + X ^ 8 :=
+begin
+  refine ((eq_cyclotomic_iff (show 0 < 15, by norm_num) _).2 _).symm,
+  rw [proper_divisors_15, finset.prod_insert _, finset.prod_insert _, finset.prod_singleton,
+    cyclotomic_one, cyclotomic_3, cyclotomic_5],
+  ring,
+  repeat { norm_num }
+end
+
+lemma cyclotomic_21 : cyclotomic 21 ℤ =
+  1 - X + X ^ 3 - X ^ 4 + X ^ 6 - X ^ 8 + X ^ 9 - X ^ 11 + X ^ 12 :=
+begin
+  refine ((eq_cyclotomic_iff (show 0 < 21, by norm_num) _).2 _).symm,
+  rw [proper_divisors_21, finset.prod_insert _, finset.prod_insert _, finset.prod_singleton,
+    cyclotomic_one, cyclotomic_3, cyclotomic_7],
+  ring,
+  repeat { norm_num }
+end
+
+lemma cyclotomic_35 : cyclotomic 35 ℤ =
+  1 - X + X ^ 5 - X ^ 6 + X ^ 7 - X ^ 8 + X ^ 10 - X ^ 11 + X ^ 12 - X ^ 13 + X ^ 14 - X ^ 16 +
+  X ^ 17 - X ^ 18 + X ^ 19 - X ^ 23 + X ^ 24 :=
+begin
+  refine ((eq_cyclotomic_iff (show 0 < 35, by norm_num) _).2 _).symm,
+  rw [proper_divisors_35, finset.prod_insert _, finset.prod_insert _, finset.prod_singleton,
+    cyclotomic_one, cyclotomic_5, cyclotomic_7],
+  ring,
+  repeat { norm_num }
+end
+
+lemma cyclotomic_105 : cyclotomic 105 ℤ =
+  1 + X + X ^ 2 - X ^ 5 - X ^ 6 - 2 * X ^ 7 - X ^ 8 - X ^ 9 + X ^ 12 + X ^ 13 + X ^ 14 + X ^ 15
+  + X ^ 16 + X ^ 17 - X ^ 20 - X ^ 22 - X ^ 24 - X ^ 26 - X ^ 28 + X ^ 31 + X ^ 32 + X ^ 33 +
+  X ^ 34 + X ^ 35 + X ^ 36 - X ^ 39 - X ^ 40 - 2 * X ^ 41 - X ^ 42 - X ^ 43 + X ^ 46 + X ^ 47 +
+  X ^ 48 :=
+begin
+  refine ((eq_cyclotomic_iff (show 0 < 105, by norm_num) _).2 _).symm,
+  rw proper_divisors_105,
+  repeat {rw finset.prod_insert _},
+  rw [finset.prod_singleton, cyclotomic_one, cyclotomic_3, cyclotomic_5, cyclotomic_7,
+    cyclotomic_15, cyclotomic_21, cyclotomic_35],
+  ring,
+  repeat { norm_num }
+end
+
+lemma coeff_cyclotomic_105 : coeff (cyclotomic 105 ℤ) 7 = -2 :=
+begin
+  have h7 : (2 * X ^ 7 : polynomial ℤ) = (C 2) * X ^ 7 := by simp,
+  have h41 : (2 * X ^ 41 : polynomial ℤ) = (C 2) * X ^ 41 := by simp,
```

Comment:
It's sad that you need to state these separately. Are we just missing a simp lemma bout `coeff`?

Reply:
I am a little confused by this: in the infoview the `2` is with no arrow (no coercion I mean), but it says `2 : polynomial ℤ`. I guess it means `1+1`, where `1` is the one of the ring. So, a lemma like
```
@[simp] lemma coeff_nmul (p : polynomial R) (a : ℕ) : coeff (a * p) n = a * coeff p n :=
```
does not apply, at least not directly. I think the relevant concept here is "numeral", but I don't know how to work with it.

Reply:
A lemma about `coeff_bit0` and `coeff_bit1` would probably help